crates/ra_hir/src/source_id.rs

```diff
@@ -115,6 +115,14 @@ impl AstIdMap {
         FileAstId { raw, _ty: PhantomData }
     }
 
+    pub(crate) fn maybe_ast_id<N: AstNode>(&self, item: &N) -> Option<FileAstId<N>> {
```

Comment:
I added this because sometimes the macro call id isn't present in the current ast id and a result it causes the server to panic.I'll need to look at why the ast_id isn't present  in the current AstIdMap

Reply:
I restarted the editor and the panics have gone away. So ill remove the method if it happens again ill make an issue